internal/provider/private_endpoint_connection_resource.go

```diff
@@ -299,10 +290,11 @@ func waitForEndpointConnectionCreatedFunc(
 		for _, *connection = range connections.GetConnections() {
 			if connection.GetEndpointId() == endpointID {
 				switch status := connection.GetStatus(); status {
-				case client.AWSENDPOINTCONNECTIONSTATUSTYPE_AVAILABLE:
+				case client.PRIVATEENDPOINTCONNECTIONSTATUS_AVAILABLE:
 					return nil
-				case client.AWSENDPOINTCONNECTIONSTATUSTYPE_PENDING,
-					client.AWSENDPOINTCONNECTIONSTATUSTYPE_PENDING_ACCEPTANCE:
+				case client.PRIVATEENDPOINTCONNECTIONSTATUS_PENDING,
+					client.PRIVATEENDPOINTCONNECTIONSTATUS_PENDING_ACCEPTANCE,
+					client.PRIVATEENDPOINTCONNECTIONSTATUS_REJECTED:
```

Comment:
the include of `REJECTED` inside the retryable error case seems wrong.  If this is actually the right thing, can you add a comment here explaining why?

Reply:
Yea seems odd but it's actually correct. Will add a comment.